src/ui/reportView/callReport.ts

```diff
@@ -114,3 +116,45 @@ function findPath(dir: string, filename: string): string {
 
 	return '';
 }
+
+/**
+ * Run the visualize command to generate the report, parse the output to return the python server command and status
+ *
+ * @param command - the cargo kani | kani command to run --visualize
+ * returns - A promise of the python command and the status code; Promise<visualizeOutput>
+ */
+async function runVisualizeCommand(command: string): Promise<visualizeOutput> {
+	try{
+		vscode.window.showWarningMessage("Generating viewer report");
+		const {stdout, stderr} = await execPromise(command);
+		const serveReportCommand = await parseReportOutput(stdout);
+		// console.error(`stderr: ${stderr}`);
+
+		return {statusCode: 0, serverCommand: serveReportCommand};
+	} catch (error) {
+		console.error(`exec error: ${error}`);
+		return {statusCode: 0, serverCommand: ''};
```

Comment:
should a caught error return a status code that does not imply success?

Reply:
No, it makes for better design if I changed the statuscode to return 1 when there is an error, I'll need to change the handling maybe for it but I can see how it's more sensible.